exporters/readers/s3_reader.py

```diff
@@ -228,88 +187,11 @@ def __init__(self, *args, **kwargs):
                                                 self.read_option('aws_access_key_id'),
                                                 self.read_option('aws_secret_access_key'))
         self.keys = self.keys_fetcher.pending_keys()
-        self.read_keys = []
-        self.current_key = None
-        self.last_block = 0
         self.logger.info('S3Reader has been initiated')
-        self.tmp_folder = tempfile.mkdtemp()
-        self.lines_reader = self.read_lines_from_keys()
 
     def get_read_streams(self):
         from exporters.bypasses.stream_bypass import Stream
         for key_name in self.keys:
             key = self.bucket.get_key(key_name)
             file_obj = urlopen(key.generate_url(S3_URL_EXPIRES_IN))
             yield Stream(file_obj, key_name, key.size)
```

Comment:
The only thing that bothers me about the stream approach (which was already changed during the migration to Kumo) is the ability to easily retry things (because the operations were more idempotent).
These changes make the S3Reader even more stateful, and it seems there is no operation really idempotent now, right?